### PR TITLE
feat: Add execute script WebDriver functions

### PR DESCRIPTION
**Pull request by Squidonomics**
This pr adds two legacy json wire protocol endpoints for executing scripts. Implements the `/session/:sessionId/execute` and `/session/:sessionId/execute_async` endpoints

### CODE REVIEW

File: Sources/WebDriver/Session.swift
Comment (compnerd) [inline Sources/WebDriver/Session.swift:76]: Does `synchronous(_:args:)` make more sense for this operation?

File: Sources/WebDriver/Session.swift
Comment (compnerd) [inline Sources/WebDriver/Session.swift:80]: Does `asynchronous(_:args:)` make more sense for this operation?

File: Sources/WebDriver/Session.swift
Comment (Squidonomics) [reply Sources/WebDriver/Session.swift:76]: I tried to follow the convention of being verbose with naming but I can make it more succinct if that would make more sense for the use

Comment (compnerd) [reply Sources/WebDriver/Session.swift:76]: Oh, that totally makes sense.  I think that we should fix the API surface to be more conventional for Swift rather than being verbose for the sake of being verbose.

Comment (jeffdav) [reply Sources/WebDriver/Session.swift]: I think we were trying to mirror the webdriver API to a certain extent, to make things intuitive (guessable).

Comment (Squidonomics) [reply Sources/WebDriver/Session.swift]: Since they're the only two script functions on the webdriver API I think making them concise for reusability makes sense.

File: Sources/WebDriver/Requests.swift
Comment (jeffdav) [inline Sources/WebDriver/Requests.swift:465]:
```suggestion
    // https://www.selenium.dev/documentation/legacy/json_wire_protocol/#sessionsessionidexecute
```

File: Sources/WebDriver/Requests.swift
Comment (jeffdav) [inline Sources/WebDriver/Requests.swift:497]:
```suggestion
                public var args: [String]
        }
    }
}
```

**Review by jeffdav — CHANGES_REQUESTED** [review]

File: Sources/WebDriver/Requests.swift
Comment (tristanlabelle) [inline Sources/WebDriver/Requests.swift:471]: Funny indentation?

File: Sources/WebDriver/Requests.swift
Comment (tristanlabelle) [inline Sources/WebDriver/Requests.swift:469]: According to spec, this could be any javascript/json value, so `Any` or `any Codable` would be a more accurate representation, if we can make this work.

File: Sources/WebDriver/Requests.swift
Comment (tristanlabelle) [inline Sources/WebDriver/Requests.swift:481]: We don't strictly need to have one struct per request url so consider just adding a `public var async: Bool` to the original struct and reusing it.

File: Sources/WebDriver/Session.swift
Comment (tristanlabelle) [inline Sources/WebDriver/Session.swift:76]: We should name this `execute` to match the protocol spec, and default to following the `script` parameter name from the spec as well.

File: Sources/WebDriver/Session.swift
Comment (tristanlabelle) [inline Sources/WebDriver/Session.swift:80]: I think it would be more ergonomic to have a single function with an `async: Bool` parameter

File: Sources/WebDriver/Session.swift
Comment (tristanlabelle) [inline Sources/WebDriver/Session.swift:76]: Double space
```suggestion
    public func synchronous(javascript script: String, args: [String]) throws {
```

File: Tests/UnitTests/APIToRequestMappingTests.swift
Comment (tristanlabelle) [inline Tests/UnitTests/APIToRequestMappingTests.swift:176]: Ideally also add a closure to assert that the request body contains the expected args.

File: Tests/UnitTests/APIToRequestMappingTests.swift
Comment (tristanlabelle) [inline Tests/UnitTests/APIToRequestMappingTests.swift:180]:
```suggestion
    func testSessionScriptAsync() throws {
```

**Review by tristanlabelle — CHANGES_REQUESTED** [review]
Thanks for your contribution! It looks good, but there's a few points where we can align with the spec better.

I'm also curious to hear whether you've been using this project to drive a web browser and how that's worked for you? We don't have such scenarios internally at The Browser Company yet.

**Comment by Squidonomics** [comment]
> Thanks for your contribution! It looks good, but there's a few points where we can align with the spec better.
> 
> I'm also curious to hear whether you've been using this project to drive a web browser and how that's worked for you? We don't have such scenarios internally at The Browser Company yet.

Thank you, always happen to create uniformity in the codebase. I don't use it for anything particular I just saw a swift project on a newsletter that came out from you guys and thought it looked cool!

File: Sources/WebDriver/Session.swift
Comment (tristanlabelle) [inline Sources/WebDriver/Session.swift:76]: Better stick with the param name from the spec, and we can make the args optional
```suggestion
    public func execute(script: String, args: [String] = [], async: Bool) throws {
```

**Review by jeffdav — APPROVED** [review]

**Comment by jeffdav** [comment]
Sorry, didn't realize this was blocked on me!

**Comment by Squidonomics** [comment]
I think this is ready to be merged @tristanlabelle

File: Tests/UnitTests/APIToRequestMappingTests.swift
Comment (compnerd) [inline Tests/UnitTests/APIToRequestMappingTests.swift:216]: Missing newline here

**Review by compnerd — APPROVED** [review]

**Comment by jeffdav** [comment]
I updated the supported APIs doc. Note that WinAppDriver doesn't support these, so there isn't a great way to test our implementation.  I suspect that we could do more to improve accessing the result / handling errors, but we can delay that until someone finds a way to use it.  :)

I think this is fine to merge once conversations are resolved.

**Comment by Squidonomics** [comment]
@jeffdav Resolved the last issue. Ready to merge.

**Pull request merged**
